ludwig/models/llm.py

```diff
@@ -467,7 +467,11 @@ def update_metrics(self, targets, predictions):
                 continue
             of_obj.update_metrics(targets[of_name], predictions[of_name])
 
+        # HACK (Tim): get the device of the targets to transfer self.eval_loss_metric to the same device
+        target_device = list(targets.values())[0].device
+
         eval_loss, additional_losses = self.eval_loss(targets, predictions)
+        self.eval_loss_metric = self.eval_loss_metric.to(target_device)
```

Comment:
Nice catch! Could you also do this for `self.eval_additional_losses_metrics`? I assume that may also require being moved to the right device (once we support multi-feature outputs for LLM model types, but might as well do it now)